logfmt: add --strict flag to enable strict parsing, perform nostrict parsing by default (#9626)
grafana/loki: docs/sources/query/log_queries/_index.md

    "fwd_ip" => "124.133.124.161"
    ```

> logfmt parser also supports `--strict` flag to enable strict parsing.

Reviewer: ```suggestion
The logfmt parser also supports using the `--strict` flag to enable strict parsing.
```

```
| logfmt --strict
| logfmt --strict host, fwd_ip="fwd"
```

> With strict parsing enabled, logfmt parser stops scanning the log line and returns early with an error if it encounters any poorly formatted key=value pair.

Reviewer: ```suggestion
With strict parsing enabled, the logfmt parser immediately stops scanning the log line and returns an error when it encounters any poorly formatted key=value pair.
```

```
// accepted
key=value key="value in double quotes" standalone_key

// invalid
=value // no key
foo=bar=buzz
fo"o=bar
```

> By default, the parser skips the invalid tokens and continues parsing the rest of the log line.

Reviewer: "By default" is unclear here, do you mean "when using the default settings" or more that skipping the invalid tokens and continuing to parse is the default behavior?

Author: @JStickler I was referring to the latter. I have updated the doc to make this clear.

Reviewer: The revision is much clearer.  Thanks!

> This offers the flexibility to parse semi-structed log lines, though note that this is only best-effort and the resulting label set might not always contain what you expect.

Reviewer: "This" is unclear, I'm not sure what you're referring to (the default behavior?)
